yt/geometry/particle_geometry_handler.py

```diff
@@ -128,13 +129,14 @@ def _initialize_index(self):
         else:
             fname = ds.index_filename
 
+        dont_load = dont_cache and not hasattr(ds, 'index_filename')
         try:
             rflag = self.regions.load_bitmasks(fname)
             rflag = self.regions.check_bitmasks()
             self._initialize_frontend_specific()
-            if rflag == 0:
+            if dont_load or rflag == 0:
                 raise OSError
-        except OSError:
+        except (OSError, struct.error):
```

Comment:
What can cause a `struct.error` here? I presume something inside `load_bitmasks`?

Reply:
Yup, if the file is mal-formed (i.e., under 8 bytes).  That shows up in the tests, for instance.